src/core/tsi/alts_transport_security.h

```diff
@@ -33,15 +33,20 @@ typedef struct alts_shared_resource {
   gpr_mu mu;
   gpr_cv cv;
   bool is_cq_drained;
+  gpr_refcount refcount;
 } alts_shared_resource;
 
-/* This method returns the address of alts_shared_resource object shared by all
- *    TSI handshakes. */
-alts_shared_resource* alts_get_shared_resource(void);
+/** This method returns mu of alts_shared_resource objet. */
+gpr_mu* grpc_tsi_alts_shared_resource_get_mu();
```

Comment:
This accessor method shouldn't be necessary, since the definition of the `alts_shared_resource` struct is externally visible.  Callers can just directly access its `mu` member.

If you do actually want to make this struct more of an opaque abstraction, then a better approach would be to move the struct definition into the .cc file (leaving only the typedef here) and add methods that allow the necessary actions to be performed without accessing the internal fields directly.  For example, the `init_shared_resources()` function in alts_tsi_handshaker.cc could be replaced with a method called `grpc_tsi_alts_shared_resource_connect()`.

Reply:
I agree the accessor method is not necessary. It is removed from the API.